src/helpers/misc.ts

```diff
@@ -212,6 +212,22 @@ export function range(start: number, end: number, step = 1) {
   return array;
 }
 
+/**
+ * Iterate trough integers from start (included) to end (included).
+ */
+export function* iterateIntegers(start: number, end: number): Generator<number> {
```

Comment:
:+1: 
I would have the same API and behavior as `range` (start, end, step) (which itself replicates the python `range` function).
Maybe even name it `rangeIterator` or something like ?

Reply:
btw, the `range` implementation could then just become `return [...rangeIterator(start, end, step)]`

Reply:
I think I'd rather make the function `range()` return an iterator, and have the dev make the spread manually (`[...range()]` if needed.

My point being that 80% of the time we only use `range()` for a `for` loop, and in this case allocating a whole array is wastefull. And there's no way peoples (me included) will think about using `rangeIterator` instead of `range` in their loops to save memory. But the otehr way around, TS will complain if we try to use an iterator as an array.